cmd/spire-agent/cli/run/run.go

```diff
@@ -354,6 +354,23 @@ func NewAgentConfig(c *Config, logOptions []log.Option, allowUnknownConfig bool)
 	serverHostPort := net.JoinHostPort(c.Agent.ServerAddress, strconv.Itoa(c.Agent.ServerPort))
 	ac.ServerAddress = fmt.Sprintf("dns:///%s", serverHostPort)
 
+	logOptions = append(logOptions,
+		log.WithLevel(c.Agent.LogLevel),
+		log.WithFormat(c.Agent.LogFormat),
+		log.WithOutputFile(c.Agent.LogFile))
+
+	logger, err := log.NewLogger(logOptions...)
+	if err != nil {
+		return nil, fmt.Errorf("could not start logger: %s", err)
+	}
+	ac.Log = logger
+
+	// Warn on a non-conforming trust domain to avoid breaking backwards compatibility
+	if len(c.Agent.TrustDomain) > common_cli.MaxTrustDomainLength {
+		logger.Warnf("Configured trust domain should be less than %v characters to be SPIFFE compliant",
```

Comment:
nit: I prefer explicit formatting options for types that have them. Then `go vet` can helpfully let you know in many cases when you have accidentally passed something wrong, e.g., a struct instead of a struct field.